drools-ha/ha-core-infra/src/test/java/org/kie/hacep/core/infra/consumer/DefaultKafkaConsumerTest.java

```diff
@@ -0,0 +1,100 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.hacep.core.infra.consumer;
+
+import java.time.LocalDateTime;
+
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.hacep.EnvConfig;
+import org.kie.hacep.consumer.DroolsConsumerHandler;
+import org.kie.hacep.core.infra.DefaultSessionSnapShooter;
+import org.kie.hacep.core.infra.SnapshotInfos;
+import org.kie.hacep.core.infra.election.State;
+import org.kie.hacep.core.infra.utils.SnapshotOnDemandUtils;
+import org.kie.hacep.util.ConsumerUtilsCore;
+import org.kie.remote.impl.producer.Producer;
+import org.kie.remote.message.ControlMessage;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DefaultKafkaConsumerTest {
+
+  @Mock
+  protected Producer producer;
+  @Mock
+  protected DroolsConsumerHandler handler;
+  @Mock
+  protected KafkaConsumer primaryConsumer;
+  @Mock
+  protected KafkaConsumer secondaryConsumer;
+  @Mock
+  protected ConsumerUtilsCore consumerUtilsCore;
+  @Mock
+  protected DefaultSessionSnapShooter defaultSessionSnapShooter;
+  @Mock
+  protected SnapshotOnDemandUtils snapshotOnDemandUtils;
+
+  private DefaultKafkaConsumer consumer;
+
+  @Before
+  public void initTest() {
+    EnvConfig envConfig = EnvConfig.getDefaultEnvConfig();
+    ControlMessage lastControlMessage = new ControlMessage();
+    lastControlMessage.setId("1");
+    lastControlMessage.setOffset(1l);
+    when(consumerUtilsCore.getLastEvent(envConfig.getControlTopicName(), envConfig.getPollTimeout())).thenReturn(lastControlMessage);
+    when(defaultSessionSnapShooter.getLastSnapshotTime()).thenReturn(LocalDateTime.now());
+    when(handler.initializeKieSessionFromSnapshotOnDemand(any(EnvConfig.class), any(SnapshotInfos.class))).thenReturn(Boolean.TRUE);
+    consumer = new DefaultKafkaConsumer(EnvConfig.getDefaultEnvConfig(), producer, primaryConsumer, secondaryConsumer, consumerUtilsCore, defaultSessionSnapShooter, snapshotOnDemandUtils, handler);
+  }
+
+  @Test
+  public void UpdateStatusBecomingLeaderAtStartupTest(){
+    InfraCallbackStatus status = consumer.updateStatus(State.BECOMING_LEADER);
+    assertFalse(status.isEnableConsumerAndStartLoop());// nothing happens
+    assertFalse(status.isUpdateOnRunningConsumer());// isn't on running consumer
+    assertFalse(status.isAskAndProcessSnapshotOnDemandResult()); // no snapshot enabled
+    assertEquals(status.getPreviousState(), State.REPLICA); // every instance starts as a replica
```

Comment:
Change to
`verify(consumer, never()).enableConsumeAndStartLoop(any());`
or
`verify(consumer, times(1)).enableConsumeAndStartLoop(eq(State.BECOMING_LEADER));`
based on the logic you want to test